sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/PredicateSuite.scala

```diff
@@ -126,6 +126,19 @@ class PredicateSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkEvaluation(IsNaN(Literal(5.5f)), false)
   }
 
+  test("NaN equality and comparison") {
+    def testNaN(nan: Expression): Unit = {
+      checkEvaluation(nan === nan, true)
+      checkEvaluation(nan <=> nan, true)
+//      checkEvaluation(nan <= nan, true)
```

Comment:
Interestingly, this test case fails even though I updated GeneratedOrdering and the interpreted orderings to support our defined NaN semantics.  This implies that we may be using the wrong ordering in the implementation of these expressions.

If it turns out that those expressions are mis-handling orderings in a more general way, then I'll open a separate PR to fix that (I suspect that we'll see similar failures when trying to order byte arrays).